### PR TITLE
Add translation

### PR DESCRIPTION
**Pull request by rayhan-ab**
add translation functionality so that we don't have to set the translation in AP

### CODE REVIEW

File: src/utils/i18n/index.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021 AccelByte Inc. All Rights Reserved
+ * This is licensed software from AccelByte Inc, for limitations
+ * and restrictions contact your company contract manager.
+ */
+
+import flatten from "flat";
```
Comment (rayhan-ab) [inline src/utils/i18n/index.ts:7]: i just copied from ap extension and deleted some unused lines

File: src/components/Card/index.tsx [hunk 2]
```diff
@@ -35,7 +36,6 @@ export interface CardProps {
   smallCard?: boolean;
   Tips?: string;
   customStyleTitleWrapper?: string;
-  noResultText?: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (rayhan-ab) [reply src/components/Card/index.tsx]: it is not deleted. this is an old commit

File: src/components/Spinner/index.tsx [hunk 3]
```diff
@@ -9,19 +9,21 @@
...
 
 export const Spinner = (props: SpinnerProps) => {
-  const { loadingText, className } = props;
+  const { loadingText = translation("loading.loading"), className, showLoadingText = true } = props;
   return (
     <div className={classNames("loader-wrapper show", className)}>
       <div className="loading-flag">
         <span className="loading-icon" />
-        {loadingText && <span className="loading-info">{loadingText}</span>}
+        {showLoadingText && <span className="loading-info">{loadingText}</span>}
```
[the remark this answers is not in the document]
Comment (rayhan-ab) [reply src/components/Spinner/index.tsx:26]: to make it the same as in AP so we can just remove the one in AP and directly import this
but maybe we can keep this component as it is and refactor AP (only 4 components using Spinner so there won't be much refactor)

File: src/utils/i18n/translations/zh-CN.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
```
[the remark this answers is not in the document]
Comment (rayhan-ab) [reply src/utils/i18n/translations/zh-CN.json:1]: updated

File: src/components/ValidatedInput/ValidSelect.tsx [hunk 5]
```diff
@@ -69,6 +69,6 @@ export const ValidSelect = ({
       isMulti={isMulti}
       dataQa={selectDataQa}
     />
-    {!optionalLabel && isInvalid && <FieldErrorMessage message={errMessage} />}
+    {isRequired && isInvalid && <FieldErrorMessage message={errMessage} />}
```
[the remark this answers is not in the document]
Comment (rayhan-ab) [reply src/components/ValidatedInput/ValidSelect.tsx]: updated by removing `isRequired`

File: src/utils/i18n/translations/en-US.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
+  "unauthorized.message": "You have no permission to view or modify it.",
+  "common.optionalFieldLabel": "(Optional)",
+  "common.add": "Add",
+  "common.refreshPage": "Refresh Page",
+  "common.more": "More",
+  "loading.loading": "Loading",
```
Comment (aziz-zen-ab) [inline src/utils/i18n/translations/en-US.json:7]: Just minor thing, does the loading component have the other translation other than `Loading`? If it doesn't have it better to use as `common.loading`, wdyt?

Comment (rayhan-ab) [reply src/utils/i18n/translations/en-US.json:7]: i actually just copied it from AP. but that sounds good too

**Review by aziz-zen-ab — APPROVED** [review]

**Pull request merged**